File: interpreter/src/pregen/scala/org/enso/interpreter/nodes/expression/operator/SubtractOperatorNode.scala
package org.enso.interpreter.nodes.expression.operator

import com.oracle.truffle.api.dsl.Specialization
import com.oracle.truffle.api.nodes.NodeInfo
import org.enso.interpreter.nodes.expression.BinaryOperatorNode

@NodeInfo(shortName = "-")
abstract class SubtractOperatorNode extends BinaryOperatorNode {

  @Specialization
  protected def subtract(left: Long, right: Long): Long =
    left - right
}

File: interpreter/src/main/java/org/enso/interpreter/node/expression/operator/SubtractOperatorNode.java
package org.enso.interpreter.node.expression.operator;

import com.oracle.truffle.api.dsl.Specialization;
import com.oracle.truffle.api.nodes.NodeInfo;
import org.enso.interpreter.node.expression.BinaryOperatorNode;

@NodeInfo(shortName = "-")
public abstract class SubtractOperatorNode extends BinaryOperatorNode {

    @Specialization
    protected long subtract(long left, long right) {
        return left - right;
    }
}
